chain-evm/src/transaction.rs

```diff
@@ -0,0 +1,49 @@
+use chain_core::{
+    packer::Codec,
+    property::{Deserialize, ReadError, Serialize, WriteError},
+};
+use ethereum::TransactionV2;
+use rlp::{decode, Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use typed_bytes::ByteBuilder;
+
+/// Wrapper type for `ethereum::TransactionV2`, which includes the `EIP1559Transaction`, `EIP2930Transaction`, `LegacyTransaction` variants.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EthereumTransaction(TransactionV2);
+
+impl EthereumTransaction {
+    /// Serialize the contract into a `ByteBuilder`.
+    pub fn serialize_in(&self, bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
+        let bytes = self.rlp_bytes();
+        bb.u64(bytes.len() as u64).bytes(&bytes)
+    }
+}
+
+impl Decodable for EthereumTransaction {
```

Comment:
I think in case that we do not want to propagate all these Ethereum traits as `Decodable` and `Encodable`, I would prefer to add just a `from_bytes(&[u8]) -> Self` and `to_bytes(self) -> &[u8]` methods for the `EthereumTransaction` it seems to be more convenient for the further usage

Reply:
I'll add these functions, although the `Serialize`/`Deserialize` traits end up providing the same, and make use of the Ethereum traits under the hood, so I think that it's equivalent.